src/composables/use-media-query.ts

```diff
@@ -49,7 +53,7 @@ export function useMediaQuery(
 
     cleanup()
 
-    mediaQuery = window.matchMedia(query)
+    mediaQuery = window.matchMedia(resolveRef(query).value)
```

Comment:
I think [Vue's built in `unref`](https://vuejs.org/api/reactivity-utilities.html#unref) makes more sense here than `resolveRef`. It seems like `resolveRef` is more useful for "make this thing a ref, one way or another" rather than a shortcut to safely getting a value out of something that may or may not be a ref :thinking: Most importantly, `unref` is a simple-ish if check and doesn't provision any new ref objects; however, `resolveRef` might actually create new ref objects which could lead to another memory leak like we had before with repeated ref creation.

Reply:
Excellent! TIL `unref` doesn't work for computed refs. Thanks for looking further into this.